components/org.wso2.carbon.identity.oauth/src/main/java/org/wso2/carbon/identity/oauth2/dao/TokenMgtDAO.java

```diff
@@ -411,8 +411,21 @@ private void storeAccessToken(String accessToken, String consumerKey, AccessToke
                     throw new IdentityOAuth2Exception(errorMsg, e);
                 }
 
-                recoverFromConAppKeyConstraintViolation(accessToken, consumerKey, accessTokenDO,
-                        connection, userStoreDomain, retryAttempt + 1);
+                boolean isPostgreSQL = false;
+                Connection pgConnection = IdentityDatabaseUtil.getDBConnection();
```

Comment:
It seems like you didn't close the pgConnection. If you did it intentionally, pls explain why?

Reply:
please check the [comment](https://github.com/wso2-support/identity-inbound-auth-oauth/pull/398#discussion_r284535906) in support PR

Reply:
In that comment it discusses about prepared statement not pgConnection. We need to close it as well.